common/components/time/time.js

```diff
@@ -0,0 +1,74 @@
+const { differenceInMinutes, isPast, parseISO } = require('date-fns')
+const timeago = require('timeago.js')
+
+function Time($module) {
+  this.cacheEls($module)
+
+  this.defaults = {
+    hasTag: this.$module.hasAttribute('data-has-tag'),
+    isRelative: this.$module.hasAttribute('data-display-relative'),
+    imminentOffset: parseInt(this.$module.getAttribute('data-imminent-offset')),
+    datetime: this.$module.getAttribute('datetime'),
+  }
+
+  this.settings = this.defaults
+}
+
+Time.prototype = {
+  init: function () {
+    if (this.settings.hasTag) {
+      this.bindEvents()
+    }
+
+    this.render()
+  },
+
+  cacheEls: function ($module) {
+    this.$module = $module
+  },
+
+  bindEvents() {
+    // Create an observer instance linked to the callback function
+    const observer = new MutationObserver(this.onChangeHandler.bind(this))
+
+    // Start observing the target node for configured mutations
+    observer.observe(this.$module, {
+      attributes: false,
+      childList: true,
+      subtree: false,
+    })
+  },
+
+  onChangeHandler(mutationsList, observer) {
+    for (let index = 0; index < mutationsList.length; index++) {
+      const mutation = mutationsList[index]
+
+      if (mutation.type === 'childList') {
+        const parsedDate = parseISO(this.settings.datetime)
+        const overdue = isPast(parsedDate)
+        const imminent =
```

Comment:
Scary variable name is scary!

Reply:
I struggled with the name of this actually...it's to define the value of when to show the yellow version of the tag...